solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/HeuristicBoostedModel.java

```diff
@@ -0,0 +1,177 @@
+package org.apache.solr.ltr.ranking;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.search.Explanation;
+import org.apache.solr.ltr.util.ModelException;
+import org.apache.solr.ltr.util.NamedParams;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This model introduces a manual boosting to the score calculation,
+ * which adds flexibility when you don't have much data for training.
+ * However, you will loose some of the benefits
+ * of a machine learned model, which was optimized to rerank your results.
+ */
+public abstract class HeuristicBoostedModel extends CompositeLTRScoringAlgorithm {
+  public static final String HEURISTIC_BOOST_PARAM = "boost";
+
+  protected EuristicFeatureBoost heuristicFeatureBoost;
+
+  public enum BoostType {product, sum};
+
+  class EuristicFeatureBoost {
+    public static final String FEATURE_NAME = "feature";
+    public static final String WEIGHT = "weight";
+    public static final String BOOST_TYPE = "type";
+
+    public String feature;
+    public int featureIndex;
+    public float weight;
+    public BoostType type;
+
+    public EuristicFeatureBoost(Map<String, Object> featureBoostParam) {
+      checkParams(featureBoostParam);
+      feature = ((String)featureBoostParam.get(FEATURE_NAME));
+      weight = NamedParams.convertToFloat(featureBoostParam.get(WEIGHT));
+      featureIndex = getFeatureIndex(feature);
+      type = BoostType.product;
+
+      if (featureBoostParam.get(BOOST_TYPE) != null) {
+        String fetchedType = (String) featureBoostParam.get(BOOST_TYPE);
```

Comment:
to convert the String to the Enum you can use `valueOf` (see `http://stackoverflow.com/questions/604424/lookup-enum-by-string-value`)